ingester/src/query.rs

```diff
@@ -89,6 +89,32 @@ impl QueryableBatch {
         }
     }
 
+    /// Return a copy of this QueryableBatch that only includes references to
+    /// snapshots with max_sequence_number > the given sequnce_number
+    pub fn copy(&self, sequence_number: Option<SequenceNumber>) -> Self {
+        let data: Vec<_> = match sequence_number {
+            None => self.data.clone(),
+            Some(seq_num) => self
+                .data
+                .clone()
+                .into_iter()
+                .filter(|s| s.max_sequencer_number > seq_num)
+                .collect(),
+        };
+
+        Self {
+            data,
+            deletes: self.deletes.clone(),
+            delete_predicates: self.delete_predicates.clone(),
+            table_name: self.table_name.clone(),
+        }
+    }
+
+    /// Add more tombstones
+    pub fn add_tombestones(&mut self, deletes: &mut Vec<Tombstone>) {
```

Comment:
Typo in method name:

```suggestion
    pub fn add_tombstones(&mut self, deletes: &mut Vec<Tombstone>) {
```